### PR TITLE
Kinematic body collision response fix (kinematic sliding bug)

### PR DESCRIPTION
**Pull request by JankMultiMediaStudio**
I patched up the kinematic body collision response to fix them sliding down slopes when the move_and_collide function is used. I did a full write up here: https://jankmultimediastudio.github.io/godot3.1-kinematic-collision-response-write-up.html

### CODE REVIEW

File: modules/bullet/space_bullet.cpp [hunk 1]
```diff
@@ -28,6 +28,9 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 
+//just for debug
+#include <iostream>
```
Comment (akien-mga) [inline modules/bullet/space_bullet.cpp]: You can use `print_line` for debug. And debug code should be removed eventually before merge.

**Comment by akien-mga** [comment]
The commits need to be squashed together, and given a proper commit title and description. See http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html and https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#be-nice-to-the-git-history

**Comment by aaronfranke** [comment]
@JankMultiMediaStudio Is this still desired? If so, it needs to be rebased on the latest master branch, and squashed together.

**Comment by AndreaCatania** [comment]
This seems the copy of this PR: https://github.com/godotengine/godot/pull/34442 Please check this comment: https://github.com/godotengine/godot/pull/34442#issuecomment-599034819

This PR will cause some other issues in other use cases.

**Comment by aaronfranke** [comment]
The source repo associated with this PR has been deleted, so this PR is no longer valid. Closing.

@JankMultiMediaStudio If this is still desired, you can open a new PR.

**Pull request closed without merging**